src/nbl/video/CVulkanCommandBuffer.cpp

```diff
@@ -479,12 +480,15 @@ bool CVulkanCommandBuffer::bindDescriptorSets_impl(asset::E_PIPELINE_BIND_POINT
 
             if (dynamicOffsets) // count dynamic offsets per set, if there are any
             {
-                auto bindings = pDescriptorSets[i]->getLayout()->getBindings();
-                for (const auto& binding : bindings)
+                auto addDynamicOffsetCount = [&dynamicOffsetCountPerSet, i](const IGPUDescriptorSetLayout::CBindingRedirect& descriptorBindingRedirect)
                 {
-                    if ((binding.type == asset::EDT_STORAGE_BUFFER_DYNAMIC) || (binding.type == asset::EDT_UNIFORM_BUFFER_DYNAMIC))
-                        dynamicOffsetCountPerSet[i] += binding.count;
-                }
+                    const auto declaredBindingCount = descriptorBindingRedirect.getBindingCount();
+                    for (uint32_t b = 0; b < declaredBindingCount; ++b)
+                        dynamicOffsetCountPerSet[i] += descriptorBindingRedirect.getCount(b);
```

Comment:
don't we have a TotalBindingCount in the `BindingRedirect` now?

Reply:
Yes, indeed, my bad, we can just use that.

Resolved.